python/ray/dataframe/dataframe.py

```diff
@@ -2308,9 +2308,43 @@ def mod(self, other, axis='columns', level=None, fill_value=None):
             "github.com/ray-project/ray.")
 
     def mode(self, axis=0, numeric_only=False):
-        raise NotImplementedError(
-            "To contribute to Pandas on Ray, please visit "
-            "github.com/ray-project/ray.")
+        axis = pd.DataFrame()._get_axis_number(axis)
+
+        def mode_helper(df):
+            if numeric_only:
+                df = df.select_dtypes(exclude='object')
+            mode_df = df.mode(axis=axis)
+            return mode_df, mode_df.shape[axis]
+
+        def reindex_helper(df, *lengths):
```

Comment:
Rename this as it's too similar to `_reindex_helper`

Reply:
resolved.